### PR TITLE
 #307 - Expose loading data loading info

### PR DESCRIPTION
**Pull request by mandragorn**
* Added 3 methods and exposed them on the DataBehavior directly and
  on the dataBehavior.data property.
** isLoading() - if anything is loading (ids or objects).
** isLoadingIds() - if ids are loading.
** isLoadingObjects() - if objects are loading from ids.

### CODE REVIEW

**Comment by mandragorn** [comment]
@kentmw - another update to peer review.

File: modules/behaviors/DataBehavior.js [hunk 1]
```diff
@@ -220,6 +220,8 @@
 
       Behavior.apply(this, arguments);
 
+      this.set('loadingIds', 0);
```
Comment (fyliu24) [inline modules/behaviors/DataBehavior.js]: Any reason for using 0 instead of true/false?

Comment (mandragorn) [reply modules/behaviors/DataBehavior.js]: yes for simultaneous requests.  If one finishes you don't want to say you are done if there is still another one in progress.

Comment (fyliu24) [reply modules/behaviors/DataBehavior.js]: oh I see

**Comment by mandragorn** [comment]
rebase

File: docs/modules/behaviors/DATA_BEHAVIOR.md [hunk 2]
```diff
@@ -808,4 +824,23 @@ successEventPayload = {
   * "behaviors.behaviorAlias:[event name]" - arbitrary event triggered by another behavior on this view.
   * "behaviors.behaviorAlias.[property name]:[event name]" - arbitrary event triggered by another behavior's property on this view.
   * "behaviors.behaviorAlias.data:[event name]" - arbitrary event triggered by another behavior's data property on this view (specific example of `behaviorAlias.[property name]:[event name]`).
-  * { '[event name]': < object (or function returning an object) that the event is triggered on > } - arbitrary "event" triggered on the supplied object.
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (kentmw) [inline docs/modules/behaviors/DATA_BEHAVIOR.md]: Merge conflicts

File: modules/behaviors/DataBehavior.js [hunk 3]
```diff
@@ -774,6 +820,36 @@
     },
 
     /**
+     * Determine if behavior is loading ids or objects.
+     * @method isLoading
+     * @return {Boolean} true - the behavior is loading objects or ids.
+     *                   false - the behavior is not loading objects or ids.
+     */
+    isLoading: function() {
```
Comment (kentmw) [inline modules/behaviors/DataBehavior.js]: just wondering how there's two isLoading methods in this file. It looks like they're different object, maybe a view and behavior? idk.

File: modules/mixins/loadingMixin.js [hunk 4]
```diff
@@ -29,28 +29,30 @@
         base.call(this, args);
         this.loadedOnceDeferred = new $.Deferred();
         this.loadedOnce = false;
+        this.loadingCount = 0;
+        // Keeping loading boolean since it is technically public and may be in use.
```
Comment (kentmw) [inline modules/mixins/loadingMixin.js]: this is a comment that makes sense in the context of this patch but it kind of confusing later on. Maybe word it like:
"loading is a convenience flag that is the equivalent of loadingCount > 0"
You may also say it's deprecated if you'd like to remove it later, though I can see its use still

File: docs/modules/behaviors/DATA_BEHAVIOR.md (hunk 2 above)
Comment (mandragorn) [reply docs/modules/behaviors/DATA_BEHAVIOR.md]: they seem to have been fixed along the way already.

File: modules/behaviors/DataBehavior.js (hunk 3 above)
Comment (mandragorn) [reply modules/behaviors/DataBehavior.js]: One is the Data object the other is the Behavior.  new (DataBehavior()).data is a full-fledged private object defined in DataBehavior.

File: modules/mixins/loadingMixin.js (hunk 4 above)
Comment (mandragorn) [reply modules/mixins/loadingMixin.js]: see #341

**Pull request merged**